lib/rules/no-duplicate-selectors/index.mjs

```diff
@@ -118,6 +125,14 @@ const rule = (primary, secondaryOptions) => {
 
 			if (!flattenedNestedSelectors.length) return;
 
+			const hasIgnoredSelector = flattenedNestedSelectors.some((selector) => {
+				const normalized = normalizeSelector(selector.clone()).toString();
+
+				return optionsMatches(secondaryOptions, 'ignoreSelectors', normalized);
+			});
+
+			if (hasIgnoredSelector) return;
```

Comment:
[suggestion] Should we skip the process if `flattenedNestedSelectors` contains an ignored selector? I guess we should filter out the ignored ones in it, instead 🤔 

E.g.

```js
const flattenedNestedSelectors = flattenNestedSelectorsForRule(ruleNode, result)
  .flatMap(({ resolvedSelectors }) => resolvedSelectors.nodes)
  .filter((selector) => !optionsMatches(secondaryOptions, 'ignoreSelectors', selector.toString()));
```